aw-server/src/endpoints/mod.rs

```diff
@@ -1,20 +1,57 @@
-use std::path::PathBuf;
+use std::ffi::OsStr;
+use std::path::{Path, PathBuf};
 use std::sync::Mutex;
 
 use gethostname::gethostname;
 use rocket::fs::FileServer;
-use rocket::fs::NamedFile;
+use rocket::http::ContentType;
 use rocket::serde::json::Json;
 use rocket::State;
 
 use crate::config::AWConfig;
 
 use aw_datastore::Datastore;
 use aw_models::Info;
+pub trait AssetResolver: Send + Sync {
+    fn resolve(&self, file_path: &str) -> Option<Vec<u8>>;
+}
+
+#[macro_export]
+macro_rules! embed_asset_resolver {
```

Comment:
Why is it necessary to have a macro for this?
Can't you instead have a single struct that includes the whole "dist" folder?

Reply:
If `__ProjectAssetResolver` is not behind the macros, then rust-embed's own derive macros always runs compile-time, which requires always having `../aw-webui/dist` relative to this rs file regardless the dynamic dispatch by trait.

Regarding the overall abstraction instead of always embedding `../aw-webui/dist`. I don't know about the strategy of this server development, hence I've made the more conservative, least prohibitive choice to allow more freedom and easier use of the server as a crate, like in the some past instances
- https://github.com/ActivityWatch/aw-tauri/blob/master/src-tauri/Cargo.toml#L19
- https://github.com/2e3s/awatcher/blob/main/Cargo.toml#L39

Reply:
Another way I thought about is `interpolate-folder-path` feature on, which allows `#[folder = "$CARGO_MANIFEST_DIR/aw-webui/dist"]` and hence this directory structure can be replicated elsewhere. I had disregarded it because it's not a maximized flexibility.

Reply:
>  Another way I thought about is interpolate-folder-path feature on, which allows #[folder = "$CARGO_MANIFEST_DIR/aw-webui/dist"] and hence this directory structure can be replicated elsewhere. I had disregarded it because it's not a maximized flexibility.

That's a pretty cool idea.

A variant on that would be to have a "AW_WEBUI_DIR" variable and if the variable is unset you can set it to "../aw-webui/dist" in build.rs.

Reply:
My initial suggestion with `interpolate-folder-path` turned out looking pretty awkward when implemented, but your improvement has made it acceptable. :+1: